### ISSUE TITLE
AudioStreamSynchronized never sets `playing` to false or emits the `finished` signal.

### ISSUE THREAD

**Issue by what-is-a-git** [godotengine/godot#89723]
### Tested versions

- Reproducible in: 4.3.dev5.official [89f70e98d]

### System information

Windows 10 - Godot 4.3.dev5.official

### Issue description

When using an AudioStreamPlayer along with the newly introduced AudioStreamSynchronized the `playing` variable is never properly set to false, and `finished` is never called when the stream stops playing. This is odd considering that, well, you'd assume you'd be told when the stream stops playing (since it **does** stop, it just doesn't tell you).

### Steps to reproduce

1. Create an AudioStreamPlayer and attach a new AudioStreamSynchronized to it.
2. Add as many streams to the AudioStreamSynchronized as you want (so far I've tested 1 stream, 3 streams of the same length, and 2 streams with different length and they all reproduced the bug).
3. Play the AudioStreamPlayer and wait for the stream to end.
4. Audio will stop playing and `get_playback_position` will return 0, however neither `playing` nor `finished` were used properly.

### Minimal reproduction project (MRP)

[audiostreamsynchronized-mrp.zip](https://github.com/godotengine/godot/files/14675871/audiostreamsynchronized-mrp.zip)

**Comment by aaronp64** [comment]
I started looking into this, seems like it's due to AudioStreamPlaybackSynchronized::mix always returning the passed in p_frames value, rather than the actual mixed frame count.  Because of this, AudioServer::_mix_step can't tell when to update that playback's state.  I tried changing it to use the max return value from each of the synchronized streams, which seems to fix it for this class.  However, it looks like AudioStreamPlaylist would have a similar issue, and maybe AudioStreamInteractive (I'm not really sure how that class is meant to be used).

Anyone more familiar with the audio code know if the above change sounds right, and how we want to handle the other classes?  I could test out my change some more and submit a PR, but I'm not sure if this should be part of a larger change.
